pkg/component/kubeapiserver/secrets.go

```diff
@@ -193,9 +194,7 @@ func (k *kubeAPIServer) reconcileSecretETCDEncryptionConfiguration(ctx context.C
 
 	encryptionConfiguration := &apiserverconfigv1.EncryptionConfiguration{
 		Resources: []apiserverconfigv1.ResourceConfiguration{{
-			Resources: []string{
-				"secrets",
-			},
+			Resources: sets.List(sets.New(k.values.ETCDEncryption.Resources...).Insert("secrets")),
```

Comment:
I would not insert `secrets` by default here. Instead, let's always populate the values field at least with `secrets` (already done in this commit).
```suggestion
			Resources: k.values.ETCDEncryption.Resources...,
```

Reply:
Why not? This is just a safety mechanism to ensure `secrets` are always present.

Reply:
Btw, I think this might change soon again anyways when the code is reused for deploying `gardener-apiserver` as part of #7016.

Reply:
I would say it's a surprising behavior of the component, as it blindly adds stuff to the given configuration.
But fine, I also doesn't really matter much.